### ISSUE TITLE
how to add background images (images to ignore)

### ISSUE THREAD

**Issue by cyberpunk-edgerunner** [roboflow/rf-detr#1186]
I want my rf-detr model to treat some images as background
will it work if I put them inside the train folder without any annotations ??

**Comment by JESUSROYETH** [comment]
Hi @cyberpunk-edgerunner 👋

Yes, background images help — RF-DETR is a DETR-style model, so it learns a "no object" prediction for every query, and feeding it images with no objects teaches it to predict no boxes on that kind of scene and lowers false positives.

But just dropping the files into the train folder is not enough, and that part is easy to miss. RF-DETR does not scan the directory, its dataloader inherits from torchvision `CocoDetection`, which iterates over the `images` array of your `_annotations.coco.json`. Any file that is not registered there is silently ignored.

So the right way is to add the image to the `images` list in the COCO json and give it no matching entries in `annotations`. The loader then returns an empty target (0 boxes, 0 labels) and the image trains as a pure background sample:

```jsonc
// _annotations.coco.json
"images": [
    {"id": 1, "file_name": "obj.jpg",        "width": 64, "height": 64},
    {"id": 2, "file_name": "background.jpg",  "width": 64, "height": 64}   // no annotations -> background
],
"annotations": [
    {"id": 1, "image_id": 1, "category_id": 1, "bbox": [10, 10, 20, 20], "area": 400, "iscrowd": 0}
]
```

I checked this on rfdetr 1.8.3. I built a dataset with 3 image files on disk but registered only 2 in the json — `len(dataset)` came back as 2 (the orphan file was dropped), and the registered image with no annotations returned `boxes=(0, 4)`, `labels=[]` with no crash ..

Versions: rfdetr 1.8.3, torch 2.12.1, torchvision 0.27.1, Python 3.12.3.

Hope this helps!

**Comment by cyberpunk-edgerunner** [comment]
That helps a lot
And I was looking at the documentation; it states that the rf-detr also
accepts annotations in YOLO format
if that is the case, can I just leave the background images in the
train/images and val/images folder without any annotations for those ones
in train/labels and val/labels ??

**Comment by JESUSROYETH** [comment]
Yes, you can. In YOLO format a background image is just an image with no matching `.txt` label file (an empty `.txt` works the same). Drop them in the train images folder without a label and rf-detr treats them as background samples with empty detections, so they push the model towards fewer false positives.

One thing to keep in mind: this used to be broken — images without a label were silently skipped — but it was fixed in #915, merged back in April. So as long as you are on a recent version (1.6 or newer, current is 1.8.3) you are fine. On anything older they just get dropped without warning ..

**Comment by JESUSROYETH** [comment]
Yes, for the YOLO format it works like you expected: you drop the background image into `train/images` (and `valid/images`) and you don't create any `.txt` for it in `labels/`. The loader then treats it as a background sample with 0 boxes.

The reason is simple: the two formats have opposite defaults, and that is easy to miss. In COCO an image that is not registered in the json gets silently ignored — that is why in my first answer you had to add it to the `images` list. In YOLO it is the other way around: RF-DETR scans `train/images` directly and, for every image, it looks for a matching `labels/<name>.txt`. If that file is missing it just returns an empty target instead of skipping the image, an empty `.txt` (0 lines) works the same way.

I checked this on rfdetr 1.8.3. I built a YOLO dataset with images in `train/images`, gave one a label file and left another with no `.txt` at all — `len(dataset)` counted both, and the unlabelled image returned `boxes=(0, 4)`, `labels=[]` with no crash:

```
background.jpg   boxes=(0, 4)  labels=[]     # no .txt   -> background
empty_lbl.jpg    boxes=(0, 4)  labels=[]     # empty .txt -> background
obj.jpg          boxes=(1, 4)  labels=[0]
```

One thing to keep in mind: the dataset still has to be a valid YOLO layout (a `data.yaml`/`data.yml`, plus `train/` and `valid/`, each with `images/` and `labels/`), otherwise RF-DETR won't detect it as YOLO in the first place.

Versions: rfdetr 1.8.3, torch 2.12.1, torchvision 0.27.1, Python 3.12.3.

Hope this helps!

**Comment by cyberpunk-edgerunner** [comment]
Thanks for your detailed explanation.

I have another question:
1. Why is RFDetr using a different number of positional encodings than
DINOv2?
Can you tell me the reason behind this? I used both large and 2XLarge, and
in both cases, it is not using the DINOv2 backbone
[2026-07-07 05:38:48] [WARNING] rf-detr - Using a different number of
positional encodings than DINOv2, which means we're not loading DINOv2
backbone weights. This is not a problem if finetuning a pretrained RF-DETR
model.
[2026-07-07 05:38:48] [WARNING] rf-detr - Using patch size 16 instead of
14, which means we're not loading DINOv2 backbone weights. This is not a
problem if finetuning a pretrained RF-DETR model.

2. Can you simplify the training output? The terminal is very verbose and
most of the time we need to use LLMs to understand the errors.

Thank you
